src/main/java/Spiderman.java

```diff
@@ -0,0 +1,212 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+import java.util.Scanner;  // Import the Scanner class
+import java.util.ArrayList; // import the ArrayList class
+
+public class Spiderman {
+    public static void main(String[] args) {
+        // Initialise scanner
+        Scanner scan = new Scanner(System.in);  // Create a Scanner object
+
+        // Initialise arrays for tasks
+        //ArrayList<Task> taskList = new ArrayList<>();
+        ArrayList<Task> taskList = loadTasksFromFile();
+
+        // Greeting users
+        System.out.println("Hello! This is your friendly neighbourhood Spiderman.");
+        System.out.println("What can I do for you?");
+
+        while (true) {
+            // Get users input
+            String input = scan.nextLine();
+            String[] splitInput = input.split("\\s+");
+            String[] splitInputByCommand = input.split("/");
+
+            // User types bye; program terminates
+            if (input.equals("bye")) {
+                break;
+            }
+
+            // User types list; display list
+            if (input.equals("list")) {
+                for (int i = 0; i < taskList.size(); i++) {
+                    System.out.println((i+1) + ". " + taskList.get(i).toString());
+                }
+                continue;
+            }
+
+            // User types delete; delete task
+            if (splitInput[0].equals("delete")) {
+                int number = Integer.parseInt(splitInput[1]) - 1;
+                System.out.println("Alright! I will delete this task for you!");
+                System.out.println(taskList.get(number).toString());
+
+                taskList.remove(number);
+                continue;
+            }
+
+            // User types mark; mark as done
+            if (splitInput[0].equals("mark")) {
+                int number = Integer.parseInt(splitInput[1]) - 1;
+                taskList.get(number).markAsDone();
+                System.out.println("Great! I've marked this task as done:");
+                System.out.println(taskList.get(number).toString());
+                continue;
+            }
+
+            // User types unmark; mark as not done
+            if (splitInput[0].equals("unmark")) {
+                int number = Integer.parseInt(splitInput[1]) - 1;
+                taskList.get(number).markAsNotDone();
+                System.out.println("OK, this task will be marked as not done yet:");
+                System.out.println(taskList.get(number).toString());
+                continue;
+            }
+
```

Comment:
The code for parsing and processing user commands is quite functional, but it could benefit from refactoring to improve readability and maintainability. The if-else blocks for different commands such as delete, mark, unmark, deadline, event, and todo could be extracted into separate methods or a command handler class. This would not only streamline the main method but also make it easier to manage and extend in the future!